staging/src/k8s.io/kube-aggregator/pkg/cmd/server/start.go

```diff
@@ -61,6 +63,9 @@ func NewCommandStartAggregator(ctx context.Context, defaults *AggregatorOptions)
 	cmd := &cobra.Command{
 		Short: "Launch a API aggregator and proxy server",
 		Long:  "Launch a API aggregator and proxy server",
+		PersistentPreRunE: func(*cobra.Command, []string) error {
+			return utilversion.DefaultComponentGlobalsRegistry.Set()
```

Comment:
This smells like it should be part of `ApplyTo` in the serverRunOptions

Reply:
this should run really right after parseFlags and before everything else, because we want to set the feature gates before any reference to it, and emulation version needs to be set before feature gates. There is no post-parse hook in pflag, the closest place is `PersistentPreRunE`. Moving it to ApplyTo could mean a lot more places where featuregate is referenced before setting the final featuregate & emulation version values.
for example: 
https://github.com/kubernetes/kubernetes/blob/9d63e575f8482b857ef328325d2107df840b9deb/staging/src/k8s.io/apiserver/pkg/server/config.go#L377 is called before [`ApplyTo`](https://github.com/kubernetes/kubernetes/blob/9d63e575f8482b857ef328325d2107df840b9deb/pkg/controlplane/apiserver/config.go#L123)
https://github.com/kubernetes/kubernetes/blob/9d63e575f8482b857ef328325d2107df840b9deb/staging/src/k8s.io/component-base/logs/api/v1/options.go#L158 is called before [`s.Complete()`](https://github.com/kubernetes/kubernetes/blob/9d63e575f8482b857ef328325d2107df840b9deb/cmd/kube-apiserver/app/server.go#L94)

Reply:
How do you plan to update the entire ecosystem to make this change to leverage the new capability here?  Perhaps it's tenable to have a backup Set in the `ApplyTo` than locate every consumer of k8s.io/apiserver.